var/spack/repos/builtin/packages/cppad/package.py

```diff
@@ -9,20 +9,32 @@
 class Cppad(CMakePackage):
     """A Package for Differentiation of C++ Algorithms."""
 
+    maintainers("whart222")
+
     homepage = "https://www.coin-or.org/CppAD/"
-    url = "http://www.coin-or.org/download/source/CppAD/cppad-20170114.gpl.tgz"
+    url = "https://github.com/coin-or/CppAD/archive/refs/tags/20240000.4.tar.gz"
     git = "https://github.com/coin-or/CppAD.git"
 
     version("develop", branch="master")
```

Comment:
The name of the version should match that of the branch (https://spack.readthedocs.io/en/latest/packaging_guide.html#git).  For one thing, it is more intuitive for people who look at the repository before building with Spack.

For more on how Spack compares versions, see https://spack.readthedocs.io/en/latest/packaging_guide.html#version-comparison.

Reply:
I made this change in the commit.